File: framework/fit/python/fit_flowable/core/exception.py
# -- encoding: utf-8 --
# Copyright (c) Huawei Technologies Co., Ltd. 2024-2024. All rights reserved.
"""
功 能：响应式框架运行错误时抛出的异常。
"""


class FlowableException(Exception):
    """
    响应式框架运行错误时抛出的异常。
    """

    def __init__(self, message: str):
        """
        初始化 FlowableException 异常。
        Args:
            message: 错误消息。
        """
        super().__init__(message)

File: framework/fit/cpp/src/runtime/exception.cpp
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2022-2022. All rights reserved.
 *
 * Description  : Provides implementation for exceptions.
 * Author       : liangjishi 00298979
 * Date         : 2022/02/24
 */

#include "include/fit/internal/exception.hpp"

using namespace ::Fit;

Exception::Exception(string message) : message_(std::move(message))
{
}

const string& Exception::GetMessage() const noexcept
{
    return message_;
}

const char* Exception::what() const noexcept
{
    return message_.c_str();
}

IllegalArgumentException::IllegalArgumentException(string message) : Exception(std::move(message))
{
}

IllegalStateException::IllegalStateException(string message) : Exception(std::move(message))
{
}